### PR TITLE
ConflictFilter 

### PR DESCRIPTION
**Pull request by kenneth-miura**
Filter that checks if any courses in the schedule have overlap in terms of time.

### CODE REVIEW

File: src/main/java/filters/ConflictFilter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import entities.Schedule;
+import entities.Section;
+import entities.Timeslot;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class ConflictFilter implements Filter {
+    private boolean allowConflicts;
+
+    public ConflictFilter(boolean allowConflicts) {
```
Comment (kenneth-miura) [inline src/main/java/filters/ConflictFilter.java:12]: Does it make sense for this filter to be able to choose to allow conflicts? Seems kinda redundant.

**Comment by sugolov** [comment]
i dont think u need the allow conflicts. if allowconflicts is true, then a schedule is always returned, so it makes no difference whether the filter is included in schedule generation or not. if user requests no conflicts, we add an instance of this filter to the filter list. otherwise, we dont.

File: src/main/java/filters/ConflictFilter.java (hunk 1 above)
Comment (sugolov) [reply src/main/java/filters/ConflictFilter.java:12]: u right

File: src/main/java/entities/Timeslot.java [hunk 2]
```diff
@@ -97,4 +97,9 @@ public boolean equals(Object o) {
         }
         return false;
     }
+
+    @Override
+    public int compareTo(Timeslot that) {
+        return this.start.compareTo(that.getStart());
```
Comment (sugolov) [inline src/main/java/entities/Timeslot.java:103]: highly clutch addition. we should merge this soon so it can be used in other places

**Pull request merged**